src/Components/Components/src/Auth/AuthorizeView.razor

```diff
@@ -54,15 +83,17 @@ else
         currentAuthenticationState = null;
 
         // Then render in completed state
+        // Importantly, we *don't* call StateHasChanged between the following async steps,
+        // otherwise we'd display an incorrect UI state while waiting for IsAuthorizedAsync
         currentAuthenticationState = await AuthenticationState;
+        isAuthorized = await IsAuthorizedAsync(currentAuthenticationState.User);
     }
 
-    private bool IsAuthorized()
+    private async Task<bool> IsAuthorizedAsync(ClaimsPrincipal user)
     {
-        // TODO: Support various authorization condition parameters, equivalent to those offered
-        // by the [Authorize] attribute, e.g., "Roles" and "Policy". This is on hold until we're
-        // able to reference the policy evaluator APIs from this package.
-
-        return currentAuthenticationState.User?.Identity?.IsAuthenticated == true;
+        var policy = await AuthorizationPolicy.CombineAsync(
+            AuthorizationPolicyProvider, selfAsAuthorizeData);
+        var result = await AuthorizationService.AuthorizeAsync(user, Resource, policy);
+        return result.Succeeded;
```

Comment:
@HaoK

Reply:
This is the same as what `DefaultHubDispatcher` does, which looks more or less equivalent in terms of its requirements and its position in the stack.

Reply:
So just to confirm I'm reading this correctly, selfAsAuthorizeData is total set of authorize attributes for this view that you want to use for the authZ check.

Reply:
Yes (except this info doesn’t come from attributes in this case- that’s a separate feature on the Blazor router).